PG-1537 Error on required arguments

AndersAstrand opened the pull request:
Do not accept NULL or the empty string for required arguments in our user facing functions.

I'm a bit unsure of where the helper `required_text_argument()` belongs, any input is appreciated.

Commit:
Require parameters in key provider functions

Previously these simply crashed the server on NULL input.

Commit:
Validate key name when creating principal key

Passing NULL here crashed the server, also do not accept the empty
string as a key name.

jeltz commented on contrib/pg_tde/src/common/pg_tde_utils.c:
```diff
@@ -94,6 +94,25 @@ get_tde_tables_count(void)
+
+	if (!PG_ARGISNULL(arg_num))
+	{
+		value = text_to_cstring(PG_GETARG_TEXT_PP(arg_num));
```
Shouldn't we have a separate error message for this, e.g. like the below?

```
# SELECT * FROM pg_create_physical_replication_slot('');
ERROR:  replication slot name "" is too short
```

AndersAstrand replied:
I'm not sure I see the value in that. But we can if we want to! :)

If we do that we should probably only do it when creating stuff and accept the empty string (and never find anything) when looking things up.

jeltz replied:
I think I do prefer that since it is after all a plugin which will be used by anyone.

AndersAstrand replied:
New version of the branch pushed, please take a look. I think this is better.

Commit:
Disallow the empty string as key provider name

Also update the error message for too long provider names to make it
symmetrical.

jeltz reviewed (approved):
Looks good!
